pulsar-metadata/src/test/java/org/apache/pulsar/metadata/CounterTest.java

```diff
@@ -51,4 +51,40 @@ public void basicTest(String provider, Supplier<String> urlSupplier) throws Exce
         long l4 = cs1.getNextCounterValue("/my/path").join();
         assertNotEquals(l3, l4);
     }
+
+    @Test(dataProvider = "impl")
+    public void testCounterDoesNotAutoReset(String provider, Supplier<String> urlSupplier) throws Exception {
+        if (provider.equals("Memory")) {
+            // Test doesn't make sense for local memory since we're testing across different instances
+            return;
+        }
+
+        MetadataStoreExtended store1 = MetadataStoreExtended.create(urlSupplier.get(),
+                MetadataStoreConfig.builder().build());
+
+        CoordinationService cs1 = new CoordinationServiceImpl(store1);
+
+        long l1 = cs1.getNextCounterValue("/my/path").join();
+        long l2 = cs1.getNextCounterValue("/my/path").join();
+        long l3 = cs1.getNextCounterValue("/my/path").join();
+
+        assertNotEquals(l1, l2);
+        assertNotEquals(l2, l3);
+
+        cs1.close();
+        store1.close();;
+
+        // Delete all the empty container nodes
+        zks.checkContainers();
+
+        MetadataStoreExtended store2 = MetadataStoreExtended.create(urlSupplier.get(),
```

Comment:
Can we validate that the znode has been really deleted?

Reply:
The point of the change is for the parent z-node to *not* be deleted :)

Reply:
Oh, right!
So we can just test that it works well